PowerNSX.psm1

```diff
@@ -21191,6 +21191,432 @@ function Remove-NsxSecurityGroupMember {
+
+    .EXAMPLE
+
+    $criteria5Spec = New-NsxDynamicCriteriaSpec -key VM.SECURITY_TAG -operator AND -criteria starts_with -value "ST_Backup"
+
+    $criteria6Spec = New-NsxDynamicCriteriaSpec -entity $(Get-Cluster Dev-CL-01) -operator AND
+
+    $criteria7Spec = New-NsxDynamicCriteriaSpec -entity $(Get-NsxLogicalSwitch LS-Backup-Net) -operator AND
+
+    $criteria8Spec = New-NsxDynamicCriteriaSpec -key VM.NAME -operator AND -criteria contains -value "PROD"
+
+    $sg3 = New-NsxSecurityGroup -Name "SG-Backup-Clients"
+
+    $sg3.objectid | Add-NsxDymanicMemberSet -operator OR -DynamicCriteriaSpec $criteria5Spec,$criteria6Spec
+    $sg3.objectid | Add-NsxDymanicMemberSet -operator OR -DynamicCriteriaSpec $criteria7Spec,$criteria8Spec
+
+    #>
+
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute("PSAvoidDefaultValueSwitchParameter","")] # Cant remove without breaking backward compatibility
+
+    param (
+        [Parameter (Mandatory=$true, ValueFromPipeline=$true, Position=1) ]
+            # SecurityGroup whose membership is to be modified.
+            [ValidateNotNullOrEmpty()]
+            [object]$SecurityGroup,
+        [Parameter (Mandatory=$true) ]
+            [ValidateSet("OR", "AND", IgnoreCase=$false)]
+            [String]$operator,
+        [Parameter (Mandatory=$true) ]
+            # Dynamic criteria spec/s as generated by New-NsxDynamicCriteriaSpec
+            [ValidateNotNullOrEmpty()]
+            [System.Xml.XmlElement[]]$DynamicCriteriaSpec,
+        [Parameter (Mandatory=$False)]
+            #PowerNSX Connection object
+            [ValidateNotNullOrEmpty()]
+            [PSCustomObject]$Connection=$defaultNSXConnection
+    )
+
+    begin {
+    }
+
+    process {
+        # Get our internal SG object and id.  The internal obejct is used later
+        # to modify and put for bulk update.
+        if ( $SecurityGroup -is [System.Xml.XmlElement] ) {
+            $SecurityGroupId = $securityGroup.objectId
+            $_SecurityGroup = $SecurityGroup.cloneNode($true)
+        }
+        elseif ( ($securityGroup -is [string]) -and ($SecurityGroup -match "securitygroup-\d+")) {
+            $SecurityGroupId = $securityGroup
+            $_SecurityGroup = Get-NsxSecurityGroup -objectId $SecurityGroupId -connection $connection
+        }
+        else {
+            throw "Invalid SecurityGroup specified.  Specify a PowerNSX SecurityGroup object or a valid securitygroup objectid."
+        }
+
+        # First we need to verify if the Security Group object passed in via the
+        # pipeline already has the dynamicMemberDefinition element created. If
+        # not, then create the required XML structure
+        if ( -not (Invoke-XPathQuery -QueryMethod SelectSingleNode -Node $_SecurityGroup -Query 'child::dynamicMemberDefinition') ) {
+            Add-XmlElement -xmlRoot $_SecurityGroup -xmlElementName "dynamicMemberDefinition"
```

Comment:
given there is no element text, you are better to just .CreateElement/AppendChild here... then you can keep the returned object rather than query for it again later....  (part1)